### PR TITLE
support specifying a secret name for ingress object TLS  

### PR DESCRIPTION
**Pull request by oferzi23**
this PR adds support for specifying a secretname in values for the TLS secret to use in the ingress object

### CODE REVIEW

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
Plz make `tls` an optional nullable property of type object, with null as default value.
The `type` property should have a single optional field `secret` which is a name of a secret to be attached to Ingress resource. If `tls` is `{}` the default secret name should be used.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
You should use conventional commits for commit message and rebase this PR to comprise a single commit.

**Comment by oferzi23** [comment]
better?

File: templates/ingress.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
   tls:
     - hosts:
         - {{ .Values.ingress.host | quote }}
-      secretName: {{ .Values.ingress.host | replace "." "-" | quote }}
+      secretName: {{ .Values.ingress.tls.secret | default (.Values.ingress.host | replace "." "-") | quote }}
```
Comment (eshepelyuk) [inline templates/ingress.yaml:20]: By default, tls is null, so here is a place for NP.E. Unfortunately, test framework is not good atm, but I believe as soon as you uncomment `tls: null` in values - the template will fail.

File: values.schema.json [hunk 2]
```diff
@@ -70,6 +70,12 @@
         "limits": { "type": "object", "title": "resource limits", "default": {} },
         "requests": { "type": "object", "title": "resource requests", "default": {} }
       }
+    },
+    "tls": {
+      "type": ["null", "object"], "title": "use TLS secret", "default": null,
+      "properties": {
+        "secret": {"type": ["null", "string"], "title": "Secret name to attach to the ingress object", "default": null }
```
Comment (eshepelyuk) [inline values.schema.json:77]: There is no need to make `secret` nullable. Since its optiinal propery - it could just b omitted.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]

File: templates/ingress.yaml (hunk 1 above)
Comment (oferzi23) [reply templates/ingress.yaml:20]: i put this in values:
```
ingress: 
  # # ingress host
  host: www.cmak.mycompany
  #
  # # ingress path
  path: /
  #
  # # Enable TLS configuration for the hostname defined at ingress.host
  # # secret name will be "${ingress.host.replace(".", "-")}"
  tls: null
  #   # secret: null
  #
  # # optional ingress annotations
  # annotations: {}
  #
  # # optional ingress labels
  # labels: {}
  ```
 and ran: ```helm template -s templates/ingress.yaml . ```
 
 i got:
 ```
 ---
# Source: cmak-operator/templates/ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: "cmak"
  labels:
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "3.0.0.5-7e7a22e"
    helm.sh/chart: "cmak-operator-0.0.0"
    app.kubernetes.io/name: "cmak"
    app.kubernetes.io/instance: "RELEASE-NAME"
spec:
  rules:
    - host: "www.cmak.mycompany"
      http:
        paths:
          - path: "/"
            backend:
              serviceName: "cmak"
              servicePort: 9000
 ```

File: values.schema.json (hunk 2 above)
Comment (oferzi23) [reply values.schema.json]: removed and left it as only string...

Comment (eshepelyuk) [reply values.schema.json]: Thnx

File: templates/ingress.yaml (hunk 1 above)
Comment (eshepelyuk) [reply templates/ingress.yaml:20]: Well, that's strange but I cant check it myself atm. Let's keep as it is.

**Review by eshepelyuk — APPROVED** [review]

**Comment by eshepelyuk** [comment]
Thanks for a contribution. I am launching a new release now.

**Comment by oferzi23** [comment]
thanks @eshepelyuk i'll keep this fork open for now as i'm working on a full implementation in my org atm. so if i come across anything else i'll just shoot a PR your way...

**Comment by eshepelyuk** [comment]
@oferzi23 
1. PRs and Features suggestions are welcomed.
1. [1.2.0](https://github.com/eshepelyuk/cmak-operator/releases/tag/1.2.0) is released , please take a look.

**Pull request merged**